File: backend/routers/analytics.py
"""
Analytics router.
"""

from typing import Literal, Optional

from fastapi import APIRouter, Depends, Query
from sqlalchemy.orm import Session

from database import get_db
from services.analytics_service import get_summary, get_timeseries

router = APIRouter(prefix="/api/analytics", tags=["analytics"])


@router.get("/summary")
def analytics_summary(
    scope: Literal["global", "regional"] = Query("global"),
    region: Optional[str] = Query(None),
    db: Session = Depends(get_db),
):
    return get_summary(db, scope=scope, region=region)


@router.get("/timeseries")
def analytics_timeseries(db: Session = Depends(get_db)):
    return {"bins": get_timeseries(db)}

File: frontend/src/api/analytics.js
export async function fetchAnalyticsSummary(scope = 'global', region) {
  const params = new URLSearchParams({ scope })
  if (scope === 'regional' && region) params.set('region', region)
  const res = await fetch(`/api/analytics/summary?${params}`)
  if (!res.ok) throw new Error(`GET /api/analytics/summary failed: ${res.status}`)
  return res.json()
}

export async function fetchAnalyticsTimeseries() {
  const res = await fetch('/api/analytics/timeseries')
  if (!res.ok) throw new Error(`GET /api/analytics/timeseries failed: ${res.status}`)
  return res.json()
}
